ecdsa/party/fullparty_test.go

```diff
@@ -502,3 +510,76 @@ func createFullParties(a *assert.Assertions, participants, threshold int, locati
 	}
 	return parties, params
 }
+
+func TestClosingThreadpool(t *testing.T) {
+	a := assert.New(t)
+
+	goroutinesstart := runtime.NumGoroutine()
+	parties, _ := createFullParties(a, test.TestParticipants, test.TestThreshold)
+
+	digestSet := make(map[Digest]bool)
+	d := crypto.Keccak256([]byte("hello, world"))
+	hash := Digest{}
+	copy(hash[:], d)
+	digestSet[hash] = false
+
+	n := networkSimulator{
+		outchan:         make(chan tss.Message, len(parties)*20),
+		sigchan:         make(chan *common.SignatureData, test.TestParticipants),
+		errchan:         make(chan *tss.Error, 1),
+		idToFullParty:   idToParty(parties),
+		digestsToVerify: digestSet,
+		Timeout:         time.Second * 3,
+		expectErr:       true,
+	}
+
+	chanReceivedAsyncTask := make(chan struct{})
+	barrier := make(chan struct{})
+	var visitedFlag int32 = 0
+	for _, p := range parties {
+		a.NoError(p.Start(n.outchan, n.sigchan, n.errchan))
+
+		tmp, ok := p.(*Impl)
+		a.True(ok)
+
+		fnc := tmp.parameters.AsyncWorkComputation
+		// setting different AsyncWorkComputation to test closing the threadpool
+		tmp.parameters.AsyncWorkComputation = func(f func()) error {
+			select {
+			// signaling we reached an async function
+			case chanReceivedAsyncTask <- struct{}{}:
+			default:
+			}
+
+			<-barrier
+			atomic.AddInt32(&visitedFlag, 1)
+			return fnc(f)
+		}
+	}
+
+	goroutinesFullThreadpools := runtime.NumGoroutine()
+	a.Greater(goroutinesFullThreadpools, goroutinesstart, "expected more goroutines")
```

Comment:
how many goroutines do you actually expect? it's better to have this assertion precise